java/edit-feature-attachments/README.md

```diff
@@ -1,24 +1,41 @@
+
+![Image of edit feature attachments](edit-feature-attachments.png)
+*****************
+## Use case
+
+Attachments provide a flexible way to manage additional information that is related to your features. Attachments allow you to add files to individual features, including: PDFs, text documents, or any other type of file. For example, if you have a feature representing a building, you could use attachments to add multiple photographs of the building taken from several angles, along with PDF files containing the building's deed and tax information.
 
 ## How to use the sample
-Tap a feature on the map to open a callout displaying number of attachments. Tap on the info button to view/edit the attachments. Selecting a list entry downloads the attachment and opens it in gallery to view. Tap on the floating action button '+' to add an attachment or long press to delete.
+
+Tap a feature on the map to open a callout displaying the number of attachments. Tap on the info button to view/edit the attachments. Select an entry from the list to download and view the attachment in the gallery. Tap on the floating action button '+' to add an attachment or long press to delete.
 
 ## How it works
-The map view provides a way to add a listener to screen taps using the `setOnTouchListener` method. The app uses the `MotionEvent` passed in to the `onSingleTapConfirmed` method to identify features on mapview based on the tolerance. `fetchAttachmentsAsync` method on selected feature returns a `List<Attachment>` list of attachments. To download an attachment `fetchDataAsync` method returns an `InputStream` which is used to download the attachment as a drawable before it can be converted to Bitmap.
 
-To add/delete an attachment, `addAttachmentAsync` and `deleteAttachmentAsync` methods are used. A listener is created to update the **ServiceFeatureTable** after finishing the async tasks of add/delete. The `updateFeatureAsync` updates the feature table and `applyEditsAsync` method is then called on the service table to apply the local changes on the server asynchronously.
+1. Create a `ServiceFeatureTable` from a URL.
+2. Create a `FeatureLayer` object from the service feature table.
+3. Select features from the feature layer with `selectFeature`.
+4. To fetch the feature's attachments, cast to an `ArcGISFeature` and use `ArcGISFeature.fetchAttachmentsAsync()`.
+5. To add an attachment to the selected ArcGISFeature, create an attachment and use `ArcGISFeature.addAttachmentAsync()`.
+6. To delete an attachment from the selected ArcGISFeature, use the `ArcGISFeature.deleteAttachmentAsync()`.
+7. After a change, apply the changes to the server using `ServiceFeatureTable.applyEditsAsync()`.
+
+## Additional information
+
+Attachments can only be added to and accessed on service feature tables when their hasAttachments property is true.
 
 ## Relevant API
-* ServiceFeatureTable
+
+* ApplyEditsAsync
+* DeleteAttachmentAsync
```

Comment:
it looks like most of these are a list of methods: but to which class do they belong?

Reply:
There's a few classes, and now I'm questioning whether the methods are necessary to include: 

```
* ArcGISFeature.deleteAttachmentAsync
* ArcGISFeature.fetchAttachmentsAsync
* Attachment.fetchDataAsync
* FeatureLayer
* ServiceFeatureTable
* ServiceFeatureTable.applyEditsAsync
* ServiceFeatureTable.UpdateFeatureAsync
```
What do you think?

Reply:
these look good to keep in: just remember to camel case the last one e.g.

ServiceFeatureTable.updateFeatureAsync

Reply:
these look good to keep in: just remember to camel case the last one e.g.

ServiceFeatureTable.updateFeatureAsync

Reply:
I've got it, thanks rachael!